core/primitives-core/src/hash.rs

```diff
@@ -74,7 +74,7 @@ impl<'de> Deserialize<'de> for CryptoHash {
 }
 
 impl std::str::FromStr for CryptoHash {
-    type Err = Box<dyn std::error::Error>;
+    type Err = Box<dyn std::error::Error + Send + Sync>;
```

Comment:
Do these need to be part of this PR? Would probably have been better to land them independently if not.

Reply:
The rest of the code depends on these. From the pov of stuff in primitives-core it is a cosmetic change, so I didn't want to create a separate PR for that.

Reply:
nvm, more stuff needs to be updated, I'll create a separate PR

Reply:
done